src/Query/Grouper.ts

```diff
@@ -1,12 +1,37 @@
 import type { Task } from '../Task';
 
 /**
- * A naming function, that takes a Task object and returns the corresponding group property name
+ * A group-naming function, that takes a Task object and returns zero or more
+ * corresponding group property names.
+ *
+ * For example, if a {@link GrouperFunction} implemented grouping by tag,
+ * passing in a task that has two tags, #tag1 and #tag2 to this tag-grouper function
+ * would return ['tag1', 'tag2']
  */
 export type GrouperFunction = (task: Task) => string[];
 
+/**
+ * A named function that is used to determine the group heading name(s) to use for a {@link Task} object.
```

Comment:
Is it correct to say that We state that the `the group heading name(s) to use` are actually `GroupDisplayHeading`s? If so, shall the `GrouperFunction` return those?

If I get it correctly the `GrouperFunction` returns the `GroupDisplayHeading.displayName`. If this is correct, it may be worth to comment this =)

Reply:
I think you are taking the detail a bit literally here... The code is a long way from being worthy of this level of scrutiny at this stage...

The main question could be 'is this code an improvement over the previous version' and maybe 'is it too big a refactoring in one go'?

Reply:
As for me this is a tremendous step towards clarity, and not much at all. I would just fix a bit things here and there (Not this one for instance).